File: Algorithms/Python/Taller02/EuclidsAlgorithm.py
class EuclidsAlgorithm():

    def FindGreatestCommonDivisor(self, FirstValue, SecondValue):
        Minor = min(FirstValue, SecondValue)
        Mayor = max(FirstValue, SecondValue)

        return self.FindGreatestCommonDivisor(Mayor, Mayor % Minor) if Mayor % Minor != 0 else Minor

def main():
    EucAlg = EuclidsAlgorithm()
    FirstValue = int(input("Set the first value: "))
    SecondValue = int(input("Set the second value: "))
    print("The Greatest Common Divisor between", FirstValue, "and", SecondValue, "is:", EucAlg.FindGreatestCommonDivisor(FirstValue, SecondValue))

if __name__ == "__main__":
    main()

File: Algorithms/Java/Taller02/EuclidsAlgorithm.java
public class EuclidsAlgorithm{
    
    public int FindGCD(int fValue, int sValue){
        int Min = Math.min(fValue, sValue);
        int Max = Math.max(fValue, sValue);
        
        if(Max%Min != 0)
        return FindGCD(Min, Max%Min);
        return Min;
    }
}